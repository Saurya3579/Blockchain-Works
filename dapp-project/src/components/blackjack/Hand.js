import React, { Component } from 'react';

import Card from './Card';

class Hand extends Component {
    static defaultProps = {
        hand: []
    }
    render() {
        return (
            <div>
                {this.props.showDeck ? <Card hidden={true} /> : ''}
                {/* here we iterate the hand array and serve the Card component with the card info */}
                {this.props.hand.map((card, i) => {
                    return <Card face={card.f} value={card.v} key={i} />
                })}
            </div>
        );
    }
}

export default Hand;
