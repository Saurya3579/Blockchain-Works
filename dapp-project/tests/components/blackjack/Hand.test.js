import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Hand from '../../../src/components/blackjack/Hand';

test('should render NoMatch component', () => {
    const wrapper = shallow(<Hand />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});