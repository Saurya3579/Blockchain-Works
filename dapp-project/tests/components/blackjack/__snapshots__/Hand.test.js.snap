exports[`test should render NoMatch component 1`] = `
<div
  className="row" />
`;
